indico/modules/events/registration/util.py

```diff
@@ -1168,3 +1169,80 @@ class CustomTicketCode:
     def lookup_registration(cls, data: str) -> Registration | None:
         """Lookup a registration based on custom ticket code data."""
         raise NotImplementedError
+
+
+def prepare_participant_list_data(reglist, display, static_items):
+    column_names = {
+        'reg_date': 'Registration Date',
+        'state': 'Registration State',
+        'price': 'Price',
+        'checked_in': 'Checked in',
+        'checked_in_date': 'Check-in Date',
+        'tags_present': 'Tags',
+    }
+
+    headers = ['ID', 'Name']
+
+    dynamic_form_items = []
+    accommodation_col_counter = 0
+
+    for item in display:
+        if item.input_type == 'accommodation':
+            accommodation_col_counter += 1
+            headers.append(item.title)
+            headers.append('Arrival date')
+            headers.append('Departure date')
+        else:
+            headers.append(item.title)
+
+        dynamic_form_items.append(item)
+
+    headers.extend(column_names[item_id] for item_id in static_items if item_id in column_names)
+
+    rows = []
+    for registration in reglist:
+        row = [f'#{registration.friendly_id}', registration.display_full_name]
+        reg_data_by_field = registration.data_by_field
+        for item in dynamic_form_items:
+            reg_data = reg_data_by_field.get(item.id)
+            friendly_data = reg_data.friendly_data if reg_data else None
+
+            if item.input_type == 'accommodation':
```

Comment:
Let's not do this in any new code.

I recently added a bunch of methods near the end of the `RegistrationFormFieldBase` class which is the base for all the fields, I think using the result from `render_reglist_column` may do the job for this since it's also a tabular list.

Reply:
https://github.com/indico/indico/blob/master/indico/modules/events/registration/templates/management/_reglist.html

I was taking this as a reference for implementing `render_reglist_column`. Aren't we using similar if/else conditions(accommodation, multiple choice etc.) here as well? Could you please explain me bit more how to implement this properly or some other reference in the codebase?

Reply:
Ugh. That's quite ugly in general... Let's put it in a single column for now like it's done in the reglist template...

Reply:
Should I move this create_pdf function to events/util.py?